zello-sdk/src/com/zello/sdk/Sdk.java

```diff
@@ -81,33 +84,35 @@ void onCreate(String packageName, Context context) {
 				@Override
 				public void onReceive(Context context, Intent intent) {
 					updateAppAvailable();
-					if (intent != null) {
-						String action = intent.getAction();
-						if (action != null) {
-							if (action.equals(Intent.ACTION_EXTERNAL_APPLICATIONS_AVAILABLE) || action.equals(Intent.ACTION_EXTERNAL_APPLICATIONS_UNAVAILABLE)) {
-								String[] pkgs = intent.getStringArrayExtra(Intent.EXTRA_CHANGED_PACKAGE_LIST);
-								if (pkgs != null) {
-									for (String pkg : pkgs) {
-										if (pkg.equalsIgnoreCase(_package)) {
-											reconnect();
-											updateSelectedContact(null);
-											updateContacts();
-											break;
-										}
-									}
-								}
-							} else {
-								Uri data = intent.getData();
-								if (data != null) {
-									String pkg = data.getSchemeSpecificPart();
-									if (pkg != null && pkg.equalsIgnoreCase(_package)) {
-										reconnect();
-										updateSelectedContact(null);
-										updateContacts();
-									}
+					if (intent == null || _handler == null) {
+						return;
+					}
+					String action = intent.getAction();
+					if (action == null) {
+						return;
+					}
+					if (action.equals(Intent.ACTION_EXTERNAL_APPLICATIONS_AVAILABLE) || action.equals(Intent.ACTION_EXTERNAL_APPLICATIONS_UNAVAILABLE)) {
+						String[] pkgs = intent.getStringArrayExtra(Intent.EXTRA_CHANGED_PACKAGE_LIST);
+						if (pkgs != null) {
+							for (String pkg : pkgs) {
+								if (pkg.equalsIgnoreCase(_package)) {
+									reconnect();
+									updateSelectedContact(null);
+									updateContacts();
+									break;
```

Comment:
`return` would be more descriptive here because that's exit point for this method

Reply:
And yes